ledger/participant-integration-api/src/main/scala/platform/store/dao/events/TransactionsWriter.scala

```diff
@@ -19,7 +19,7 @@ import com.daml.lf.transaction.BlindingInfo
 import com.daml.metrics.{Metrics, Timed}
 import com.daml.platform.store.DbType
 
-private[dao] object TransactionsWriter {
```

Comment:
By no means ideal, but exposing the `PreparedInsert` to the JdbcIndexer would have otherwise meant extracting a big pile of stuff from `dao`. I decided against that massive refactoring since this part will be revisited heavily in the near future. What do you think @stefano

Reply:
That it's fine, but we should at least have a ticket for that. My fear is that it's one of those tasks we'll never deal with.

Reply:
Ok, would you have a Github issue or a JIRA ticket?

Reply:
@gerolf-da What do you think in this case? Is it fine to create a ticket for this ? (With a summary something like _[Refactoring] Separate indexer entry preparation from DAO/SQL execution_

Reply:
This is needed for testing then?

Reply:
@gerolf-da This is needed in order to allow executing the transaction prepare in the `JdbcIndexer` flow (outside the `LedgerDao` DAO)

Reply:
I believe this takes care of the whole thing by just making `LedgerDao`'s interface wider, which is a sin we're already guilty of. :trollface: